runtime/jsMain/src/kotlinx/serialization/DynamicObjectSerializer.kt

```diff
@@ -0,0 +1,182 @@
+package kotlinx.serialization
+
+import kotlinx.serialization.internal.StringDescriptor
+import kotlinx.serialization.internal.nullable
+import kotlinx.serialization.json.internal.BEGIN_LIST
+import kotlinx.serialization.json.internal.BEGIN_OBJ
+import kotlinx.serialization.json.internal.END_LIST
+import kotlinx.serialization.json.internal.END_OBJ
+import kotlinx.serialization.modules.EmptyModule
+import kotlinx.serialization.modules.SerialModule
+import kotlinx.serialization.modules.getContextualOrDefault
+
+
+/**
+ * Converts Kotlin data structures to plain Javascript objects
+ *
+ *
+ * Limitations:
+ * * Map keys must be of primitive or enum type
+ * * Custom serializers are not supported
+ * * Enums are serialized as the value of `@SerialName` if present or their name, in that order. Ordinals are not supported.
+ * * There is no analogue to DynamicObjectParsers' useArrayPolymorphism flag
```

Comment:
I'd rephrase this to 'currently does not support polymorphism'. It seems it doesn't work even with array polymorphism now. I have some thoughts how to fix that, so you can leave that to me.